datafusion/src/physical_plan/windows.rs

```diff
@@ -248,21 +252,25 @@ impl AggregateWindowExpr {
         let values = self.evaluate_args(batch)?;
         let results = partition_points
             .iter()
-            .map(|partition_range| {
+            .map::<Result<_>, _>(|partition_range| {
                 let sort_partition_points =
                     find_ranges_in_range(partition_range, &sort_partition_points);
                 let mut window_accumulators = self.create_accumulator()?;
-                sort_partition_points
+                let result = sort_partition_points
                     .iter()
                     .map(|range| window_accumulators.scan_peers(&values, range))
-                    .collect::<Result<Vec<_>>>()
+                    .collect::<Result<Vec<_>>>()?
+                    .into_iter()
+                    .flatten();
+                ScalarValue::iter_to_array(result)
```

Comment:
@Dandandan i find that changing to use this, even if would be saving ArrayRef creating, makes the performance much worse, because it changes from iterating trusted len iterator to this boxed type with unbounded untrusted iterator.